### PR TITLE
Feature - KhuyenDang - CustomView

### PR DESCRIPTION
**Pull request by mvn-khuyendang-dn**

### CODE REVIEW

File: 05_CustomView/KhuyenDang/CustomView/CustomView/CustomView/Bai1/UserViewController.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+class UserViewController: UIViewController {
+    
+    @IBOutlet weak var scrollView: UIScrollView!
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        var xUser: CGFloat = 15
+        var yUser: CGFloat = 50
+        let widthUser: CGFloat = CGFloat((UIScreen.main.bounds.width - 15 * 4) / 3) //122
+        for _ in 0...9 {
```
Comment (mvn-tamnguyen7-dn) [inline 05_CustomView/KhuyenDang/CustomView/CustomView/CustomView/Bai1/UserViewController.swift:18]: please define 15 , 4

File: 05_CustomView/KhuyenDang/CustomView/CustomView/CustomView/Bai1/View/UserView.swift [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+//  Created by Khuyen Dang T.T. VN.Danang on 6/20/22.
+//
+
+import UIKit
+
+protocol UserViewDelegate: AnyObject {
+    func didTaped(view: UserView, name: String)
+}
+
+class UserView: UIView {
```
Comment (mvn-tamnguyen7-dn) [inline 05_CustomView/KhuyenDang/CustomView/CustomView/CustomView/Bai1/View/UserView.swift:14]: final

File: 05_CustomView/KhuyenDang/CustomView/CustomView/CustomView/Bai1/View/UserView.swift [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    weak var delegate: UserViewDelegate?
```
Comment (mvn-tamnguyen7-dn) [inline 05_CustomView/KhuyenDang/CustomView/CustomView/CustomView/Bai1/View/UserView.swift:15]: line space

File: 05_CustomView/KhuyenDang/CustomView/CustomView/CustomView/Bai1/View/UserView.swift [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        button.backgroundColor = .clear
+        button.addTarget(self, action: #selector(buttonDidClick), for: .touchUpInside)
+        self.addSubview(button)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    @objc func buttonDidClick() {
```
Comment (mvn-tamnguyen7-dn) [inline 05_CustomView/KhuyenDang/CustomView/CustomView/CustomView/Bai1/View/UserView.swift:45]: @objc private func

File: 05_CustomView/KhuyenDang/CustomView/CustomView/CustomView/HomeVC/HomeViewController.swift [hunk 5]
```diff
@@ -0,0 +1,56 @@
+//
+//  HomeViewController.swift
+//  Demo
+//
+//  Created by Tien Le P. VN.Danang on 6/16/22.
+//
+
+import UIKit
+
+class HomeViewController: UIViewController {
```
Comment (mvn-tamnguyen7-dn) [inline 05_CustomView/KhuyenDang/CustomView/CustomView/CustomView/HomeVC/HomeViewController.swift]: class nào k kế thừa thì final hết nha em.

File: 05_CustomView/KhuyenDang/CustomView/CustomView/CustomView/HomeVC/HomeViewController.swift [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        
+        // 4. AddSubView
+        //    UIView nào đó --> 2 điều:
+        //       - Mình có thể add nó vào một View khác (View cha)
+        //       - Mình có thể add cho nó nhiều view con (subviews)
+        self.view.addSubview(myView)
+        
+        
+    }
+
```
Comment (mvn-tamnguyen7-dn) [inline 05_CustomView/KhuyenDang/CustomView/CustomView/CustomView/HomeVC/HomeViewController.swift]: delete khoảng trống dư

File: 05_CustomView/KhuyenDang/CustomView/CustomView/CustomView/Bai4/DatePickerViewController.swift [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+//
+
+import UIKit
+
+final class DatePickerViewController: UIViewController {
+    
+    @IBOutlet private weak var button: UIButton!
+    @IBOutlet private weak var dateTextField: UITextField!
+    
+    weak var myDatePicker: MyDatePickerView!
```
Comment (mvn-tanhuynh-dn) [inline 05_CustomView/KhuyenDang/CustomView/CustomView/CustomView/Bai4/DatePickerViewController.swift:15]: Please don't use force to unwrap(!)

File: 05_CustomView/KhuyenDang/CustomView/CustomView/CustomView/Bai4/DatePickerViewController.swift [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        myDatePicker = UINib(nibName: "MyDatePickerView", bundle: .main).instantiate(withOwner: self).first as? MyDatePickerView
+        myDatePicker.frame = UIScreen.main.bounds
+        self.view.addSubview(myDatePicker)
```
Comment (mvn-tanhuynh-dn) [inline 05_CustomView/KhuyenDang/CustomView/CustomView/CustomView/Bai4/DatePickerViewController.swift:21]: Don't need self

**Review by mvn-tanhuynh-dn — APPROVED** [review]
LGTM!

**Pull request merged**